auth-manifest/app/src/main.rs

```diff
@@ -0,0 +1,130 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+   main.rs
+
+Abstract:
+
+    Main entry point for Caliptra Authorization Manifest application
+
+--*/
+
+use anyhow::Context;
+use caliptra_auth_man_gen::{AuthManifestGenerator, AuthManifestGeneratorConfig};
+#[cfg(feature = "openssl")]
+use caliptra_image_crypto::OsslCrypto as Crypto;
+#[cfg(feature = "rustcrypto")]
+use caliptra_image_crypto::RustCrypto as Crypto;
+use clap::ArgMatches;
+use clap::{arg, value_parser, Command};
+use std::io::Write;
+use std::path::PathBuf;
+use zerocopy::AsBytes;
+
+mod config;
+
+/// Entry point
+fn main() {
+    let sub_cmds = vec![Command::new("create-auth-man")
+        .about("Create a new authorization manifest")
+        .arg(
+            arg!(--"version" <U32> "Manifest Version Number")
+                .required(true)
+                .value_parser(value_parser!(u32)),
+        )
+        .arg(
+            arg!(--"flags" <U32> "Manifest Flags")
+                .required(true)
+                .value_parser(value_parser!(u32)),
+        )
+        .arg(
+            arg!(--"key-dir" <FILE> "Key files directory path")
+                .required(true)
+                .value_parser(value_parser!(PathBuf)),
+        )
+        .arg(
+            arg!(--"config" <FILE> "Configuration file")
+                .required(true)
+                .value_parser(value_parser!(PathBuf)),
+        )
+        .arg(
+            arg!(--"key-dir" <FILE> "Key files directory path")
+                .required(true)
+                .value_parser(value_parser!(PathBuf)),
+        )
+        .arg(
+            arg!(--"out" <FILE> "Output file")
+                .required(true)
+                .value_parser(value_parser!(PathBuf)),
+        )];
+
+    let cmd = Command::new("caliptra-auth-man-app")
+        .arg_required_else_help(true)
+        .subcommands(sub_cmds)
+        .about("Caliptra authorization manifest tools")
+        .get_matches();
+
+    let result = match cmd.subcommand().unwrap() {
+        ("create-auth-man", args) => run_auth_man_cmd(args),
+        (_, _) => unreachable!(),
+    };
+
+    result.unwrap();
+}
+
+pub(crate) fn run_auth_man_cmd(args: &ArgMatches) -> anyhow::Result<()> {
+    let version: &u32 = args
+        .get_one::<u32>("version")
+        .with_context(|| "version arg not specified")?;
+
+    let flags: &u32 = args
```

Comment:
nit: Could use `bitflags` crate throughout these libraries to make the code a little more readable.

Reply:
Updated.